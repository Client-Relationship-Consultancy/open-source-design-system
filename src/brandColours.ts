export const colourPalette = {
  examplePalette: {
    primary: {
      main: {
        hex: "#A42824",
        RGB: "164,40,36",
        on: "#FFFFFF",
      },
      light: {
        hex: "#DA342F",
        RGB: "218,52,47",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#661614",
        RGB: "102,22,20",
        on: "#FFFFFF",
      },
    },
    secondary: {
      main: {
        hex: "#0B466B",
        RGB: "11,70,107",
        on: "#FFFFFF",
      },
      light: {
        hex: "#599CAF",
        RGB: "89,156,175",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#05263B",
        RGB: "5,38,59",
        on: "#FFFFFF",
      },
    },
    complimentary: {
      main: {
        hex: "#FFEE52",
        RGB: "255,238,82",
        on: "#272727",
      },
      light: {
        hex: "#FFF59D",
        RGB: "255,245,157",
        on: "#272727",
      },
      dark: {
        hex: "#CEB900",
        RGB: "206,185,0",
        on: "#272727",
      },
    },
    action: {
      main: {
        hex: "#743C8F",
        RGB: "116,60,143",
        on: "#FFFFFF",
      },
      light: {
        hex: "#AD65CF",
        RGB: "173,101,207",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#4E2662",
        RGB: "79,38,98",
        on: "#FFFFFF",
      },
    },
    error: {
      main: {
        hex: "#FF7320",
        RGB: "255,115,32",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#C54901",
        RGB: "197,73,1",
        on: "#FFFFFF",
      },
    },
    background: {
      hex: "#F6F6F6",
      RGB: "246,246,246",
    },
    surface: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    white: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    black: {
      main: { hex: "#272727", RGB: "39,39,39", on: "#FFFFFF" },
      tint80: { hex: "#696969", RGB: "105,105,105", on: "#FFFFFF" },
      tint60: { hex: "#A0A0A0", RGB: "160,160,160", on: "#FFFFFF" },
      tint40: { hex: "#C7C7C7", RGB: "199,199,199", on: "#272727" },
      tint20: { hex: "#ECECEC", RGB: "236,236,236", on: "#272727" },
    },
  },
};

interface IColourCodes {
  hex: string;
  RGB: string;
  on?: string;
}
interface IColourVariations {
  main: IColourCodes;
  light?: IColourCodes;
  dark?: IColourCodes;
}

interface IBlackVariations {
  main: IColourCodes;
  tint80: IColourCodes;
  tint60: IColourCodes;
  tint40: IColourCodes;
  tint20: IColourCodes;
}

export interface IColourPalette {
  primary: IColourVariations;
  secondary: IColourVariations;
  complimentary: IColourVariations;
  action: IColourVariations;
  error: IColourVariations;
  background: IColourCodes;
  surface: IColourCodes;
  white: IColourCodes;
  black: IBlackVariations;
}
