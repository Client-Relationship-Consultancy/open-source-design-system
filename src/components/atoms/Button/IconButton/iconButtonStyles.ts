import { IColourPalette } from "../../../../brandColours";

export type IconButtonStyle = {
  background: string;
  color: string;
  border: string;
  hover?: {
    background?: string;
    color?: string;
    border?: string;
  };
  focus?: {
    background?: string;
    color?: string;
    border?: string;
  };
};

export const iconButtonStyles = (theme: IColourPalette) => ({
  primary: {
    background: theme.action.main.hex,
    color: theme.action.main.on,
    border: theme.action.main.hex,
    hover: {
      background: theme.action.dark?.hex,
      color: theme.action.dark?.on,
      border: theme.action.dark?.hex,
    },
    focus: {},
  },
  primaryOutline: {
    background: "transparent",
    color: theme.action.main.hex,
    border: theme.action.main.hex,
    hover: {
      background: theme.action.dark?.hex,
      color: theme.action.dark?.on,
      border: theme.action.dark?.hex,
    },
    focus: {
      background: theme.action.main.hex,
      border: theme.action.main.hex,
      color: theme.action.main.on,
    },
  },
  secondaryOutline: {
    background: "transparent",
    color: theme.secondary.main.hex,
    border: theme.secondary.main.hex,
    hover: {
      background: theme.secondary.dark?.hex,
      color: theme.secondary.dark?.on,
      border: theme.secondary.dark?.hex,
    },
    focus: {
      background: theme.secondary.main.hex,
      color: theme.secondary.main.on,
      border: theme.secondary.main.hex,
    },
  },
  danger: {
    background: theme.error.main.hex,
    color: theme.error.main.on,
    border: theme.error.main.hex,
    hover: {
      background: theme.error.dark?.hex,
      color: theme.error.dark?.on,
      border: theme.error.dark?.hex,
    },
    focus: {
      background: theme.error.main.hex,
      color: theme.error.main.on,
      border: theme.error.main.hex,
    },
  },
  ghost: {
    background: "transparent",
    color: theme.action.main.hex,
    border: "transparent",
    hover: {
      background: theme.background.hex,
    },
    focus: {
      background: theme.background.hex,
      border: theme.action.main.hex,
    },
  },
});
