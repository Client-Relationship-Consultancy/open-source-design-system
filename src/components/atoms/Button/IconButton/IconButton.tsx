import React from "react";
import styled, { withTheme } from "styled-components";
import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { colourPalette, IColourPalette } from "../../../../brandColours";
import Icon from "../../Icon/Icon";
import { iconButtonStyles, IconButtonStyle } from "./iconButtonStyles";

export type ButtonType = "primary" | "primaryOutline" | "secondaryOutline" | "danger" | "ghost";
export type IconType = IconProp | SVGElement | React.ReactElement;
export type IconPosition = "before" | "after";
export type ButtonSize = "small" | "medium" | "large";

interface IStyledIconButton {
  buttonStyle: IconButtonStyle;
  buttonDimensions: string;
  iconDimensions: string;
  buttonSize: ButtonSize;
}

export const StyledIconButton = styled.button<IStyledIconButton>`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  outline: none;
  height: ${({ buttonDimensions }) => buttonDimensions};
  width: ${({ buttonDimensions }) => buttonDimensions};
  background-color: ${({ buttonStyle }) => buttonStyle.background};
  color: ${({ buttonStyle }) => buttonStyle.color};
  border: 1px solid ${({ buttonStyle }) => buttonStyle.border};
  border-radius: 50%;
  transition-property: color, background-color, border;
  transition-duration: 0.15s;
  .dsButtonIcon {
    width: ${({ iconDimensions }) => iconDimensions};
    height: ${({ iconDimensions }) => iconDimensions};
    display: flex;
    align-items: center;
    svg {
      height: ${({ iconDimensions }) => iconDimensions};
      width: ${({ iconDimensions }) => iconDimensions};
      color: ${({ buttonStyle }) => buttonStyle.color};
      fill: ${({ buttonStyle }) => buttonStyle.color};
      stroke: ${({ buttonStyle }) => buttonStyle.color};
    }
  }
  :enabled {
    &:hover {
      cursor: pointer;
      background-color: ${({ buttonStyle }) =>
        buttonStyle.hover?.background ?? buttonStyle.background};
      color: ${({ buttonStyle }) => buttonStyle.hover?.color ?? buttonStyle.color};
      border: 1px solid ${({ buttonStyle }) => buttonStyle.hover?.border ?? buttonStyle.border};
    }
    &:active,
    &:focus {
      background-color: ${({ buttonStyle }) =>
        buttonStyle.focus?.background ?? buttonStyle.background};
      color: ${({ buttonStyle }) => buttonStyle.focus?.color ?? buttonStyle.color};
      border: 1px solid ${({ buttonStyle }) => buttonStyle.focus?.border ?? buttonStyle.border};
    }
    &:hover,
    &:active {
      svg {
        color: ${({ buttonStyle }) => buttonStyle.hover?.color ?? buttonStyle.color};
        fill: ${({ buttonStyle }) => buttonStyle.hover?.color ?? buttonStyle.color};
        stroke: ${({ buttonStyle }) => buttonStyle.hover?.color ?? buttonStyle.color};
      }
    }
    &:focus {
      svg {
        color: ${({ buttonStyle }) => buttonStyle.focus?.color ?? buttonStyle.color};
        fill: ${({ buttonStyle }) => buttonStyle.focus?.color ?? buttonStyle.color};
        stroke: ${({ buttonStyle }) => buttonStyle.focus?.color ?? buttonStyle.color};
      }
    }
  }

  :disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
`;

StyledIconButton.displayName = "StyledIconButton";

interface IProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  buttonSize?: ButtonSize;
  buttonType?: ButtonType;
  theme: IColourPalette;
  disabled?: boolean;
  icon?: IconType;
}

class BasicIconButton extends React.PureComponent<IProps> {
  static defaultProps = {
    theme: colourPalette.examplePalette,
    buttonType: "primary",
  };

  getButtonDimensions = (): string => {
    switch (this.props.buttonSize) {
      case "large":
        return "2.5rem";
      case "medium":
        return "2rem";
      case "small":
        return "1.5rem";
      default:
        return "2rem";
    }
  };

  getIconDimensions = (): string => {
    switch (this.props.buttonSize) {
      case "large":
        return "1.25rem";
      case "medium":
        return "1.125rem";
      case "small":
        return "1rem";
      default:
        return "1.125rem";
    }
  };

  getButtonStyle = (): IconButtonStyle =>
    iconButtonStyles(this.props.theme)[this.props.buttonType ?? "primary"];

  render = () => {
    const { disabled = false, buttonSize = "medium", onClick, icon } = this.props;

    return (
      <StyledIconButton
        {...this.props}
        buttonStyle={this.getButtonStyle()}
        buttonDimensions={this.getButtonDimensions()}
        iconDimensions={this.getIconDimensions()}
        onClick={onClick}
        disabled={disabled}
        buttonSize={buttonSize}
      >
        {icon &&
          (typeof icon === "string" ? (
            <Icon name={icon} className="dsButtonIcon" />
          ) : (
            <div className="dsButtonIcon">{icon}</div> // wrapper for SVG or react element
          ))}
      </StyledIconButton>
    );
  };
}

const IconButton = withTheme(BasicIconButton);
IconButton.displayName = "IconButton";

export default IconButton;
