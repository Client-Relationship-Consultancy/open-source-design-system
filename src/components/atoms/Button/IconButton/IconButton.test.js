import React from "react"
import { mount } from "enzyme"
import "jest-styled-components"
import { colourPalette } from "../../../../brandColours"

import IconButton from "./IconButton"

const { action } = colourPalette.examplePalette

describe("IconButton", () => {
  it("should match snapshot with defaults", () => {
    const component = mount(<IconButton />)
    expect(component).toMatchSnapshot()
  })

  it("should pass the correct icon and button dimension props to StyledIconButton based on buttonSize", () => {
    const smallButton = mount(<IconButton buttonSize="small" />)
    expect(smallButton.find("StyledIconButton").props().buttonDimensions).toEqual("1.5rem")
    expect(smallButton.find("StyledIconButton").props().iconDimensions).toEqual("1rem")

    const mediumButton = mount(<IconButton buttonSize="medium" />)
    expect(mediumButton.find("StyledIconButton").props().buttonDimensions).toEqual("2rem")
    expect(mediumButton.find("StyledIconButton").props().iconDimensions).toEqual("1.125rem")

    const largeButton = mount(<IconButton buttonSize="large" />)
    expect(largeButton.find("StyledIconButton").props().buttonDimensions).toEqual("2.5rem")
    expect(largeButton.find("StyledIconButton").props().iconDimensions).toEqual("1.25rem")
  })

  it("should render the icon if it's passed in as a non-string (svg) value", () => {
    const selectArrowSvg = (
      <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
        <circle cx="50" cy="50" r="40" strokeWidth="5" />
      </svg>
    )

    const component = mount(<IconButton icon={selectArrowSvg} />)

    expect(component.find(".dsButtonIcon")).toHaveLength(1)
    expect(component.contains(selectArrowSvg)).toEqual(true)
  })

  it("should have the correct ghost styling when passed ghost prop", () => {
    const component = mount(<IconButton buttonType="ghost" />)

    expect(component.find("StyledIconButton")).toHaveStyleRule("color", action.main.hex)
    expect(component.find("StyledIconButton")).toHaveStyleRule("background-color", "transparent")
    expect(component.find("StyledIconButton")).toHaveStyleRule("border", "1px solid transparent")
  })

  it("should not render text child when passed", () => {
    const component = mount(<IconButton>Button Label</IconButton>)

    expect(component.find("span")).toHaveLength(0)
    expect(component).not.toContain("Button Label")
  })
})
