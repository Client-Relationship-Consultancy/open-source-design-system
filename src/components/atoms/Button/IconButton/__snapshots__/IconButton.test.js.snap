// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot with defaults 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  outline: none;
  height: 2rem;
  width: 2rem;
  background-color: #743C8F;
  color: #FFFFFF;
  border: 1px solid #743C8F;
  border-radius: 50%;
  -webkit-transition-property: color,background-color,border;
  transition-property: color,background-color,border;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.c0 .dsButtonIcon {
  width: 1.125rem;
  height: 1.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .dsButtonIcon svg {
  height: 1.125rem;
  width: 1.125rem;
  color: #FFFFFF;
  fill: #FFFFFF;
  stroke: #FFFFFF;
}

.c0:enabled:hover {
  cursor: pointer;
  background-color: #4E2662;
  color: #FFFFFF;
  border: 1px solid #4E2662;
}

.c0:enabled:active,
.c0:enabled:focus {
  background-color: #743C8F;
  color: #FFFFFF;
  border: 1px solid #743C8F;
}

.c0:enabled:hover svg,
.c0:enabled:active svg {
  color: #FFFFFF;
  fill: #FFFFFF;
  stroke: #FFFFFF;
}

.c0:enabled:focus svg {
  color: #FFFFFF;
  fill: #FFFFFF;
  stroke: #FFFFFF;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<IconButton>
  <BasicIconButton
    buttonType="primary"
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
          "on": "#272727",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <StyledIconButton
      buttonDimensions="2rem"
      buttonSize="medium"
      buttonStyle={
        Object {
          "background": "#743C8F",
          "border": "#743C8F",
          "color": "#FFFFFF",
          "focus": Object {},
          "hover": Object {
            "background": "#4E2662",
            "border": "#4E2662",
            "color": "#FFFFFF",
          },
        }
      }
      buttonType="primary"
      disabled={false}
      iconDimensions="1.125rem"
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
            "on": "#272727",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
    >
      <StyledComponent
        buttonDimensions="2rem"
        buttonSize="medium"
        buttonStyle={
          Object {
            "background": "#743C8F",
            "border": "#743C8F",
            "color": "#FFFFFF",
            "focus": Object {},
            "hover": Object {
              "background": "#4E2662",
              "border": "#4E2662",
              "color": "#FFFFFF",
            },
          }
        }
        buttonType="primary"
        disabled={false}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "IconButton__StyledIconButton-sc-1a6m2r2-0",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "display:flex;align-items:center;justify-content:center;position:relative;outline:none;height:",
                [Function],
                ";width:",
                [Function],
                ";background-color:",
                [Function],
                ";color:",
                [Function],
                ";border:1px solid ",
                [Function],
                ";border-radius:50%;transition-property:color,background-color,border;transition-duration:0.15s;.dsButtonIcon{width:",
                [Function],
                ";height:",
                [Function],
                ";display:flex;align-items:center;svg{height:",
                [Function],
                ";width:",
                [Function],
                ";color:",
                [Function],
                ";fill:",
                [Function],
                ";stroke:",
                [Function],
                ";}}:enabled{&:hover{cursor:pointer;background-color:",
                [Function],
                ";color:",
                [Function],
                ";border:1px solid ",
                [Function],
                ";}&:active,&:focus{background-color:",
                [Function],
                ";color:",
                [Function],
                ";border:1px solid ",
                [Function],
                ";}&:hover,&:active{svg{color:",
                [Function],
                ";fill:",
                [Function],
                ";stroke:",
                [Function],
                ";}}&:focus{svg{color:",
                [Function],
                ";fill:",
                [Function],
                ";stroke:",
                [Function],
                ";}}}:disabled{opacity:0.5;cursor:not-allowed;}",
              ],
            },
            "displayName": "StyledIconButton",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "IconButton__StyledIconButton-sc-1a6m2r2-0",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        iconDimensions="1.125rem"
        theme={
          Object {
            "action": Object {
              "dark": Object {
                "RGB": "79,38,98",
                "hex": "#4E2662",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "173,101,207",
                "hex": "#AD65CF",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "116,60,143",
                "hex": "#743C8F",
                "on": "#FFFFFF",
              },
            },
            "background": Object {
              "RGB": "246,246,246",
              "hex": "#F6F6F6",
              "on": "#272727",
            },
            "black": Object {
              "main": Object {
                "RGB": "39,39,39",
                "hex": "#272727",
                "on": "#FFFFFF",
              },
              "tint20": Object {
                "RGB": "236,236,236",
                "hex": "#ECECEC",
                "on": "#272727",
              },
              "tint40": Object {
                "RGB": "199,199,199",
                "hex": "#C7C7C7",
                "on": "#272727",
              },
              "tint60": Object {
                "RGB": "160,160,160",
                "hex": "#A0A0A0",
                "on": "#FFFFFF",
              },
              "tint80": Object {
                "RGB": "105,105,105",
                "hex": "#696969",
                "on": "#FFFFFF",
              },
            },
            "complimentary": Object {
              "dark": Object {
                "RGB": "206,185,0",
                "hex": "#CEB900",
                "on": "#272727",
              },
              "light": Object {
                "RGB": "255,245,157",
                "hex": "#FFF59D",
                "on": "#272727",
              },
              "main": Object {
                "RGB": "255,238,82",
                "hex": "#FFEE52",
                "on": "#272727",
              },
            },
            "error": Object {
              "dark": Object {
                "RGB": "197,73,1",
                "hex": "#C54901",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "255,115,32",
                "hex": "#FF7320",
                "on": "#FFFFFF",
              },
            },
            "primary": Object {
              "dark": Object {
                "RGB": "102,22,20",
                "hex": "#661614",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "218,52,47",
                "hex": "#DA342F",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "164,40,36",
                "hex": "#A42824",
                "on": "#FFFFFF",
              },
            },
            "secondary": Object {
              "dark": Object {
                "RGB": "5,38,59",
                "hex": "#05263B",
                "on": "#FFFFFF",
              },
              "light": Object {
                "RGB": "89,156,175",
                "hex": "#599CAF",
                "on": "#FFFFFF",
              },
              "main": Object {
                "RGB": "11,70,107",
                "hex": "#0B466B",
                "on": "#FFFFFF",
              },
            },
            "surface": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
            "white": Object {
              "RGB": "255,255,255",
              "hex": "#FFFFFF",
              "on": "#272727",
            },
          }
        }
      >
        <button
          className="c0"
          disabled={false}
        />
      </StyledComponent>
    </StyledIconButton>
  </BasicIconButton>
</IconButton>
`;
